### ISSUE TITLE
Error "getornull: Condition !id_map.has(p_rid.get_data())" showing up when changing property of a sprite 3D with a deleted light source

### ISSUE THREAD

**Issue by Pantabrah** [godotengine/godot#48118]
The error message :

```
getornull: Condition "!id_map.has(p_rid.get_data())" is true. Returned: __null
   At: ./core/rid.h:160 @ getornull()
```

Is displayed everytime you change a property of a sprite3D (Like the current frame or the opacity) that has been touched by a light source that no longer exist until another light source touch it.
Even if the sprite is out of range of the light source when it get removed, the moment the light stop existing the error will begin showing up.
Only spotlights or omnilights trigger the problem.

Tested on 3.3-stable as well as on a 3.2.4RC1 that was laying around.

Here is a small project displaying the issue :

[test_light.zip](https://github.com/godotengine/godot/files/6361615/test_light.zip)

**Comment by Calinou** [comment]
Can you reproduce this after disabling BVH in the Project Settings?

**Comment by Pantabrah** [comment]
Yes, disabling BVH does not solve the issue.

**Comment by akien-mga** [comment]
> Tested on the lastest stable release as well as on a 3.2.4RC1 that was laying around.

Please clarify what "latest stable release" means, in a few weeks time this will be different from what you mean today and contributors shouldn't have to look up the chronology of releases to infer this information.

**Comment by akien-mga** [comment]
Tested a few releases, this appears to be a regression in 3.2.3-stable (compared to 3.2.2-stable which does not produce the error).

Probably related to #39867.

**Comment by clayjohn** [comment]
Error is coming from this line:
https://github.com/godotengine/godot/blob/48d7eff3e36b4e0b9e15266d2a84d32c1e43db93/drivers/gles3/rasterizer_scene_gles3.cpp#L1756

Still can't figure out why Sprite3D is triggering the error.

**Comment by akien-mga** [comment]
Fixed by #48499.
